src/components/Input/Input.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useCallback, useState, useEffect } from "react"
+import { BigNumber, ethers } from "ethers"
+
+import styles from "./Input.module.scss"
+
+type InputToken = "SHER" | "USDC"
+
+type InputProps = {
+  /**
+   * onChange event handler
+   */
+  onChange?: (value: BigNumber | null) => void
+
+  /**
+   * Token
+   */
+  token: InputToken
+
+  /**
+   * Placeholder
+   */
+  placeholder?: string
+}
+
+const decimalsByToken: Record<InputToken, number> = {
+  SHER: 18,
+  USDC: 6,
+}
+
+const decommify = (value: string) => value.replaceAll(",", "")
+
+export const Input: React.FC<InputProps> = ({ onChange, token, placeholder }) => {
+  const [stringValue, setStringValue] = useState<string>("")
```

Comment:
Can we use here the `useAmountState` hook to hold the amount?

Reply:
yeah, good call